packages/jest-cli/src/TestRunner.js

```diff
@@ -34,6 +42,7 @@ class TestRunner {
 
     utils.createDirectory(this._config.cacheDirectory);
 
+    config.moduleFileExtensions.push('snap');
```

Comment:
I'd prefer to add it here: https://github.com/facebook/jest/blob/master/packages/jest-cli/src/lib/createHasteMap.js#L20 without mutating the config option that Jest has (ie. use .slice there). Also, let's export the extension from `jest-snapshot` and reuse it here so we don't have more than one place where `snap` is mentioned directly.